recipes/libpng/all/conanfile.py

```diff
@@ -96,8 +96,7 @@ def package_info(self):
                 self.cpp_info.libs = ['libpng16']
         else:
             self.cpp_info.libs = ["png16"]
-            if self.settings.os == "Linux":
-                self.cpp_info.libs.append("m")
+            self.cpp_info.libs.append("m")
```

Comment:
Probably is safer to keep the "IF" adding android, otherwise other NIX systems might suffer.